test/test_distributions.py

```diff
@@ -4092,5 +4092,191 @@ def tearDown(self):
+                Laplace,  # aten::uniform(Double(5, 5), float, float, Generator)
+                LogNormal,  # aten::normal(Double(5, 5), float, float, Generator)
+                LogisticNormal,  # aten::normal(Double(5, 5), float, float, Generator)
+                LowRankMultivariateNormal,  # aten::normal(Dynamic, float, float, Generator)
+                MultivariateNormal,  # aten::normal(Dynamic, float, float, Generator)
+                Normal,  # aten::normal(Double(5, 5), float, float, Generator)
+                OneHotCategorical,  # aten::scatter(Double(2, 3), int, Dynamic, int)
+                Pareto,  # aten::exponential(Double(5, 5), float, Generator)
+                RelaxedBernoulli,  # aten::uniform(Double(3), float, float, Generator)
+                RelaxedOneHotCategorical,  # aten::uniform(Double(2, 3), float, float, Generator)
+                StudentT,  # aten::normal(Double(2, 3), float, float, Generator)
+                Uniform,  # aten::uniform(Double(5, 5), float, float, Generator)
+                Weibull,  # aten::exponential(Double(5, 5), float, Generator)
+            ]
+            if Dist in xfail:
+                print('        xfail reproducibility')
+                continue
+
+            with torch.random.fork_rng():
+                sample_0 = traced_f(*values)
+            sample_1 = traced_f(*values)
+            self.assertEqual(sample_0, sample_1)
+
+            # FIXME no nondeterministic nodes found in trace
+            xfail = [Beta, Dirichlet]
+            if Dist in xfail:
+                print('        xfail nondeterministic nodes')
+                continue
+            self.assertTrue(any(n.isNondeterministic() for n in traced_f.graph.nodes()))
+
+    def test_log_prob(self):
+        for Dist, keys, values, sample in self._examples():
+
+            def f(sample, *values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.log_prob(sample)
+
+            torch.jit.trace(f, (sample,) + values)
+
+    def test_enumerate_support(self):
+        for Dist, keys, values, sample in self._examples():
+            if not Dist.has_enumerate_support:
+                continue
+
+            def f(*values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.enumerate_support()
+
+            try:
+                torch.jit.trace(f, values)
+            except NotImplementedError:
+                continue
+
+    def test_mean(self):
+        for Dist, keys, values, sample in self._examples():
+
+            def f(*values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.mean
+
+            try:
+                torch.jit.trace(f, values)
+            except NotImplementedError:
+                continue
+
+    def test_variance(self):
+        for Dist, keys, values, sample in self._examples():
```

Comment:
Is it possible to add a small test for distribution transforms, i.e. transform a sample to unconstrained space and back again to constrained space? I noticed that some of our transforms are using in-place operations like `copy_`, so it will be good to know the failure cases.

Reply:
Done.